keps/0007-kic-performance-test.md

```diff
@@ -0,0 +1,557 @@
+---
+title: KIC Performance Test Solution
+status: implementable
+---
+
+# Notes
+For references see the dependencies related to this proprosal to check the progress of related efforts:
+- solution 
+- deployment and configuration
+- code/repo
+- configuration and execution
+- profiling (code profiler)
+- histogram execution time and calculates percentiles
+
+
+## Summary
+  We want a performance test solution that can collect controller metrics per release, which also provides profiling
+and histogram execution time. The solution is able to scale according to controller scalability. We mainly use kind as test cluster, but also support varied cloud provider cluster such EKS, AKS, GKE etc.
+
+[kic]:https://github.com/kong/kubernetes-ingress-controller
+[ktf]:https://github.com/Kong/kubernetes-testing-framework
+[jenkins]: https://www.jenkins.io/
+[aws]: https://aws.amazon.com/
+[kind]: https://kind.sigs.k8s.io/
+[docker]: https://www.docker.com/ 
+
+## Motivation
+  - provide an isolated and reproducible environment for performance testing
+  - continue integration and continue test
+  - Visualize testing configuration and results.
+  - System distribution
```

Comment:
Generally the motivation section should be causal factors, whereas these all appear to be more like goals in that they're specific things we want to achieve.

I would like to suggest the following alternative for motivations:

```suggestion
- we currently have no profiles, documentation, or determined performance characteristics for the KIC in any Kubernetes or infrastructure environment
- we want libraries and tooling that can be run against a Kubernetes cluster and its underlying infrastructure to analyze the performance characteristics of the KIC in that environment
- we want our tooling to be open source so end-users can run a performance analysis themselves in their environment
```

Based on the history of this proposal that I'm aware of, I think that this covers some of the motivations that originally led to this project earlier in the year. Let me know your thoughts?

Reply:
it looks like the opposite way presenting the same thing, 
this PR - want to do A,B,C
You are suggesting: we do not have A, B, C

I do not have strong preference, would like to follow your thought.

Reply:
Right, "we do not have A, B, C" provides the motivators. The "motivation" are the things that are currently true that have lead us to want to propose a change.